### PR TITLE
docs: add note about MAC addresses not being unique

### LINKED ISSUE #22174: Docker allows you to launch 2 containers with the same mac address
In window 1

```
docker run -it --rm --mac-address 00:11:22:33:44:55 ubuntu /bin/bash
```

In window 2

```
docker run -it --rm --mac-address 00:11:22:33:44:55 ubuntu /bin/bash
```

Actual result: 

Both containers launch, both have the same mac address. 

Desired result: 

Second container does not launch and instead returns the error:

```
Container "raging_bob" is already configured to use 00:11:22:33:44:55
```

### PR DESCRIPTION
**Pull request by thaJeztah**
fixes https://github.com/docker/docker/issues/22174

### CODE REVIEW

File: docs/reference/run.md [hunk 1]
```diff
@@ -293,6 +293,10 @@ By default, the MAC address is generated using the IP address allocated to the
 container. You can set the container's MAC address explicitly by providing a
 MAC address via the `--mac-address` parameter (format:`12:34:56:78:9a:bc`).
 
+Be aware that Docker does not check if manually specified MAC addresses are
+unique. This is by design, because a container can be connected to a different
+L2 domain.
```
Comment (aboch) [inline docs/reference/run.md]: I would end after `unique.` I think that is the only info needed by the user who wants to play with setting the container's MAC address. Reason why docker does not monitor uniqueness of user chosen mac address may be several, from technical to philosophical or simply practical and not relevant to the user guide context, in my opinion.

Comment (thaJeztah) [inline docs/reference/run.md]: updated :)

**Comment by aboch** [comment]
LGTM

**Comment by SvenDowideit** [comment]
LGTM

**Pull request merged**